apps/hubble/src/storage/engine/storageCache.ts

```
@@ -0,0 +1,101 @@
+import {
+  HubEvent,
+  isMergeMessageHubEvent,
+  isPruneMessageHubEvent,
+  isRevokeMessageHubEvent,
+  Message,
+} from '@farcaster/protobufs';
+import { ok, err } from 'neverthrow';
+import { HubError, HubResult } from '~/../../../packages/utils/dist';
+import RocksDB from '~/storage/db/rocksdb';
+import { FID_BYTES, RootPrefix, UserMessagePostfix, UserMessagePostfixMax } from '~/storage/db/types';
+import { logger } from '~/utils/logger';
+import { makeFidKey, typeToSetPostfix } from '../db/message';
+
+const makeKey = (fid: number, set: UserMessagePostfix): string => {
+  return Buffer.concat([makeFidKey(fid), Buffer.from([set])]).toString('hex');
+};
+
+const log = logger.child({ component: 'StorageCache' });
+
+export class StorageCache {
+  private _usage: Map<string, number>;
+  private _synced = false;
+
+  constructor(usage?: Map<string, number>) {
+    this._usage = usage ?? new Map();
+  }
+
+  async syncFromDb(db: RocksDB): Promise<void> {
+    log.info('starting storage cache sync');
+    const usage = new Map<string, number>();
+
+    const prefix = Buffer.from([RootPrefix.User]);
+
+    const iterator = db.iteratorByPrefix(prefix);
+
+    for await (const [key] of iterator) {
+      const postfix = (key as Buffer).readUint8(1 + FID_BYTES);
+      if (postfix < UserMessagePostfixMax) {
+        const usageKey = (key as Buffer).subarray(1, 1 + FID_BYTES + 1).toString('hex');
+        const count = usage.get(usageKey) ?? 0;
+        usage.set(usageKey, count + 1);
+      }
+    }
+
+    this._usage = usage;
+    this._synced = true;
+    log.info('storage cache synced');
+  }
+
+  getMessageCount(fid: number, set: UserMessagePostfix): HubResult<number> {
+    if (this._synced !== true) {
+      return err(new HubError('unavailable.storage_failure', 'storage cache is not synced with db'));
```

Consider dropping a log statement here since the callers ignore the error